opmd_viewer/addons/pic/lpa_diagnostics.py

```diff
@@ -17,4 +20,668 @@ def __init__(self, path_to_dir):
+            an available file, the last file before `t` will be used)
+            Either `t` or `iteration` should be given by the user.
+
+        iteration : int
+            The iteration at which to obtain the data
+            Either `t` or `iteration` should be given by the user.
+
+        pol : float
+            Polarization angel of the field relative to the x plane. Can be
+            freely chosen between 0 and Pi/2 in thetaMode. For carthesian
+            coordinates it as to be either 0 or Pi/2 (x or y plane)
+
+        Returns
+        -------
+        - A 2d array with wigner transform
+        - extents (time centered around the laser pulse in s and the
+          angular frequency)
+        """
+        # Test that polarization is in x or y plane for 3D cart. coordinates
+        if self.geometry == "3dcartesian" or self.geometry == "2dcartesian":
+            if pol == 0:
+                slicing_dir = 'y'
+                coord = 'x'
+            elif pol == np.pi/2:
+                slicing_dir = 'x'
+                coord = 'y'
+            else:
+                raise ValueError('Only polarization in the x or y plane is '
+                                 'supported for carthesian coordinates')
+        else:
+            # Extract radial if in thetaMode
+            coord = 'r'
+            slicing_dir = None
+        # Get the field envelope
+        env, _ = self.get_laser_envelope(t=t, iteration=iteration,
+                                         pol=pol)
+        # Get the field
+        E, extent = self.get_field( t=t, iteration=iteration, field='E',
+                                    coord=coord, theta=pol,
+                                    slicing_dir=slicing_dir )
+        # Get central slice
+        E = E[E.shape[0] / 2, :]
+        # Get time domain of the data
+        tmin = extent.zmin / const.c
+        tmax = extent.zmax / const.c
+        T = tmax - tmin
+        dt = T / E.size
+        # Normalize the Envelope
+        env /= np.sqrt(np.trapz(env ** 2, dx=dt))
+        # Allocate array for shifted E field and wigner transform
+        E_shift = np.zeros_like(E)
+        wigner = np.zeros((2 * E.size, E.size))
+        # Construct shifted E field
+        for i in range(E.size * 2):
+            itau = i % E.size
+            if i < E.size:
+                E_shift[:itau] = env[E.size - itau: E.size]
+                E_shift[itau:] = 0
+            else:
+                E_shift[itau:] = env[: E.size - itau]
+                E_shift[:itau] = 0
+            EE = E * E_shift ** 2
+            fftwigner = np.fft.fft(EE)
+            wigner[i, :] = np.abs(fftwigner) ** 2
+        # Rotate and flip array to have input form of imshow
+        wigner = np.flipud(np.rot90(wigner[:, E.size / 2:]))
+        # Calculate the axis range
+        maxi, maxj = np.unravel_index(wigner.argmax(), wigner.shape)
+        xmin = -(T - T / wigner.shape[1] * maxj)
+        xmax = -(0 - T / wigner.shape[1] * maxj)
```

Comment:
At this point, instead of the above lines
```python
 maxi, maxj = np.unravel_index(wigner.argmax(), wigner.shape)
 xmin = -(T - T / wigner.shape[1] * maxj)
 xmax = -(0 - T / wigner.shape[1] * maxj)
```
I would initialize a FieldMetaInformation object:
```
info = FieldMetaInformation( {0:'t', 1:'omega'}, wigner.shape,
         gridspacing=(dt, np.pi/dt/wigner.shape), grid_unitSI=1.,
        global_offset=(0,0), position=(0,0))
return( wigner, info )
```
but the above is most certainly wrong: I am not sure my `gridspacing` is correct, and my `global_offset` (the start of the grid along each axis) is certainly wrong. Do you think you could adapt the above to make it right ?